manifest/schema2/manifest.go

```diff
@@ -48,6 +52,14 @@ func init() {
 	}
 }
 
+// LayerDescriptor is an extension to distribution.Descriptor that
+// has additional fields specific to layers.
+type LayerDescriptor struct {
+	distribution.Descriptor
+	// ForeignURL contains the foreign source URL of this layer.
+	ForeignURL string `json:"foreign_url,omitempty"`
```

Comment:
Would you expect a client to try to try each of the URLs in order until one succeeded?